test/OpenTelemetry.Extensions.Tests/Trace/BaggageActivityProcessorTests.cs

```diff
@@ -0,0 +1,115 @@
+            .AddSource(sourceName)
+            .Build();
+
+        Baggage.SetBaggage("key", "value");
+        Baggage.SetBaggage("other_key", "other_value");
+
+        using var source = new ActivitySource(sourceName);
+        using var activity = source.StartActivity("name", ActivityKind.Server);
+        Assert.NotNull(activity);
+        activity.Stop();
+
+        Assert.Contains(activity.Tags, kv => kv.Key == "key" && kv.Value == "value");
+        Assert.Contains(activity.Tags, kv => kv.Key == "other_key" && kv.Value == "other_value");
+    }
+
+    [Fact]
+    public void BaggageActivityProcessor_CanUseCustomPredicate()
+    {
+        var activityProcessor = new TestActivityProcessor();
+        var sourceName = GetTestMethodName();
+
+        using var provider = Sdk.CreateTracerProviderBuilder()
+            .AddProcessor(activityProcessor)
+            .AddBaggageActivityProcessor((baggageKey) => baggageKey.StartsWith("key", StringComparison.Ordinal))
+            .AddSource(sourceName)
+            .Build();
+
+        Baggage.SetBaggage("key", "value");
+        Baggage.SetBaggage("other_key", "other_value");
+
+        using var source = new ActivitySource(sourceName);
+        using var activity = source.StartActivity("name", ActivityKind.Client);
+        Assert.NotNull(activity);
+        activity.Stop();
+
+        Assert.Contains(activity.Tags, kv => kv.Key == "key" && kv.Value == "value");
+        Assert.DoesNotContain(activity.Tags, kv => kv.Key == "other_key" && kv.Value == "other_value");
+    }
+
+    [Fact]
+    public void BaggageActivityProcessor_CanUseRegex()
+    {
+        var activityProcessor = new TestActivityProcessor();
+        var sourceName = GetTestMethodName();
+
+        var regex = new Regex("^mykey", RegexOptions.Compiled);
+        using var provider = Sdk.CreateTracerProviderBuilder()
+            .AddProcessor(activityProcessor)
+            .AddBaggageActivityProcessor((baggageKey) => regex.IsMatch(baggageKey))
+            .AddSource(sourceName)
+            .Build();
+
+        Baggage.SetBaggage("mykey", "value");
+        Baggage.SetBaggage("other_key", "other_value");
+
+        using var source = new ActivitySource(sourceName);
+        using var activity = source.StartActivity("name", ActivityKind.Client);
+        Assert.NotNull(activity);
+        activity.Stop();
+
+        Assert.Contains(activity.Tags, kv => kv.Key == "mykey" && kv.Value == "value");
+        Assert.DoesNotContain(activity.Tags, kv => kv.Key == "other_key" && kv.Value == "other_value");
+    }
+
+    [Fact]
+    public void BaggageActivityProcessor_PredicateThrows_DoesNothing()
+    {
+        var activityProcessor = new TestActivityProcessor();
+        var sourceName = GetTestMethodName();
+
+        using var provider = Sdk.CreateTracerProviderBuilder()
+            .AddProcessor(activityProcessor)
+            .AddBaggageActivityProcessor(_ => throw new Exception("Predicate throws an exception."))
```

Comment:
I'd suggest a test to validate the behavior - when an exception is thrown, is entire baggage dropped? or just that particular baggage entry is dropped? If we are dropping only the particular baggage entry, then the error message should contain the keyname.

Reply:
Added extra test to show only entries that throw are dropped in [325f39f](https://github.com/open-telemetry/opentelemetry-dotnet-contrib/pull/1816/commits/325f39f0836150f163c088451fe13de2b69ba37f)